src/pack_unpack.c

```diff
@@ -440,6 +440,7 @@ static int unpack_object(scanner_t *s, json_t *root, va_list *ap)
         strbuffer_t unrecognized_keys;
         json_t *value;
         long unpacked = 0;
+        strbuffer_init(&unrecognized_keys);
```

Comment:
It's a performance optimization to initialize `unrecognized_keys` only if it's needed.